### ISSUE TITLE
Add ability to grab focus on PopupMenu items

### ISSUE THREAD

**Issue by ShatReal** [godotengine/godot-proposals#2663]
### Describe the project you are working on
RPG with menus navigable by keyboard

### Describe the problem or limitation you are having in your project
When PopupMenu opens, no item is selected. This is confusing for the user since they have to press up or down to select an item.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation
If an item can grab focus this confusion can be avoided.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams
Make some kind of function that will allow an item to grab focus.

### If this enhancement will not be used often, can it be worked around with a few lines of script?
No, you need to use a different node.

### Is there a reason why this should be core and not an add-on in the asset library?
PopupMenu is core.

**Comment by EricEzaM** [comment]
When you right click or open context menus on windows, no item is selected by default. 

Or did you just want a function like `popup_menu->highlight_item(int id)`

**Comment by ShatReal** [comment]
Yeah I want a function like that. I want to be able to highlight an item when the menu pops up for people using the keyboard to navigate menus.

**Comment by rambda** [comment]
A workaround to this:
```
func set_current_index(index: int):
	show()
	grab_focus()
	
	var action: String
	var offset := index - get_current_index()
	if offset > 0:
		action = "ui_down"
	else:
		action = "ui_up"
	
	var e := InputEventAction.new()
	e.action = action
	e.pressed = true
	for _i in abs(offset):
		Input.parse_input_event(e)
```
